hs_core/views/utils.py

```diff
@@ -1100,7 +1106,19 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
                     if aggregation_object:
                         aggregation_object.logical_delete(user)
                     else:
-                        istorage.delete(override_tgt_path)
+                        # check if destination override path is a file
+                        f = ResourceFile.get(resource=resource, file=override_tgt_path)
+                        if f:
+                            f.delete()
+                        else:
+                            # check if destination override path is a folder
+                            files = ResourceFile.objects.filter(resource=resource,
+                                                                file_folder=override_tgt_path.rsplit('data/contents/'))
+                            if files:
+                                # delete all files in the folder
+                                files.delete()
+                            # delete the folder in iRODS
+                            istorage.delete(override_tgt_path)
```

Comment:
I think this line which deletes the folder will also delete all files in that folder. In the `else` block only this line should be sufficient. Also we need to use `remove_folder` function from `views/utils.py` to delete the folder which will take care of any aggregations associated with that folder.

Reply:
Good point. Thanks for the suggestion.